cpp/ql/src/experimental/Security/CWE/CWE-476/DangerousUseOfExceptionBlocks.cpp

```diff
@@ -0,0 +1,48 @@
+...
+try {
+  if (chackValue) throw exception();
+  valData->bufMyData =  new myData*[valData->sizeInt];
+	 
+  } 
+  catch (...) 
+  {
+    for (size_t i = 0; i < valData->sizeInt; i++)
+    {
+    	delete[] valData->bufMyData[i]->buffer; // BAD
+        delete valData->bufMyData[i];
+    }
+...
+try {
+  if (chackValue) throw exception();
+  valData->bufMyData =  new myData*[valData->sizeInt];
+	 
+  } 
+  catch (...) 
+  {
+    for (size_t i = 0; i < valData->sizeInt; i++)
+    {
+      if(delete valData->bufMyData[i])
+      {
+    	  delete[] valData->bufMyData[i]->buffer; // GOOD
+        delete valData->bufMyData[i];
+      }
+    }
+
+...
+  catch (const exception &) {
+	  delete valData;
+	  throw;
+  }
+  catch (...) 
+  {
+    delete valData; // BAD
+...
+  catch (const exception &) {
+	  delete valData;
+    valData = NULL;
+	  throw;
+  }
+  catch (...) 
+  {
+    delete valData; // GOOD  
+...
```

Comment:
I struggled to follow these examples.  I feel like they (and most of the tests) could probably contain fewer levels of pointer indirection and still get the same meaning across.

Reply:
I agree with you.
but this is a forced complexity.
I need to repeat difficulty level CVE.

Reply:
Its good to have simple and complicated cases in the tests.  However in the example, we need to be as simple as possible so that the reader can immediately see what's going on.

Reply:
I understand.
I will definitely make the tests and examples simpler.

Reply:
good afternoon @geoffw0.
I tried to simplify.
Please take a look, maybe something else needs to be simplified.

Reply:
can I make this item resolved?

Reply:
Yeah, I think so.